### ISSUE TITLE
Lmod settarg errors on a read-only/fusemount filesystem

### ISSUE THREAD

**Issue by fgeorgatos** [TACC/Lmod#347]
hi @rtmclay ,

i got the following bug report yesterday, any hints on how to collect more information about it?

```
(we) keep having LUA errors on read-only filesystems (for example /onefusemount, that's why they reported it to me), because lmod tries to do something in there:
 
/usr/bin/lua: /usr/share/lmod/lmod/settarg/BuildTarget.lua:284: bad argument #1 to 'load' (string expected, got nil)
stack traceback:
[C]: in function 'load'
/usr/share/lmod/lmod/settarg/BuildTarget.lua:284: in function 'readDotFiles'
/usr/share/lmod/lmod/settarg/BuildTarget.lua:358: in function 'exec'
/usr/share/lmod/lmod/settarg/settarg_cmd:157: in function 'main'
/usr/share/lmod/lmod/settarg/settarg_cmd:183: in main chunk
[C]: ?

They get this error as soon as they `cd` into a r/o fusemount directory: There's something that gets automatically executed as part of their PROMPT...
 
Would you know more about it, and how we can avoid having this error?
 ```

Although the faint-hearted approach would be to stand down `settarg` from the default list of modules, I would rather understand why the above occurs and do something better with it.

any comments?

**Comment by fgeorgatos** [comment]
Hi @rtmclay ,

thanks for looking into this; it might still be something fancy on me end, but is this reproducible for you with the latest version(s)?

```
[fgeorgatos@node005 ~]$ ml --version

Modules based on Lua: Version 7.7.20  2018-03-14 12:36 -05:00
    by Robert McLay mclay@tacc.utexas.edu

[fgeorgatos@node005 ~]$ mkdir posix.lua
/usr/bin/lua: error loading module 'posix' from file './posix.lua':
        cannot read ./posix.lua: Is a directory
stack traceback:
        [C]: ?
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/../tools/fileOps.lua:35: in main chunk
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/../tools/serializeTbl.lua:47: in main chunk
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/STT.lua:41: in main chunk
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/BaseShell.lua:41: in main chunk
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/settarg_cmd:76: in main chunk
        [C]: ?
[fgeorgatos@node005 ~]$ date
Tue Mar 20 07:50:55 PDT 2018
/usr/bin/lua: error loading module 'posix' from file './posix.lua':
        cannot read ./posix.lua: Is a directory
stack traceback:
        [C]: ?
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/../tools/fileOps.lua:35: in main chunk
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/../tools/serializeTbl.lua:47: in main chunk
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/STT.lua:41: in main chunk
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/BaseShell.lua:41: in main chunk
        [C]: in function 'require'
        /usr/share/lmod/lmod/settarg/settarg_cmd:76: in main chunk
        [C]: ?
```

same story with `.settarg.lua` - or we claim so ;-)

**Comment by rtmclay** [comment]
You have Lmod configured wrong.  What does ml --config say.  In particular what is LUA_PATH?

**Comment by fgeorgatos** [comment]
spot on:
```
$ ml --config 2>&1 |grep LUA_PATH
LUA_PATH                         ./?.lua;/usr/share/lua/5.1/?.lua;/usr/share/lua/5.1/?/init.lua;/usr/lib64/lua/5.1/?.lua;/usr/lib64/lua/5.1/?/init.lua
```

I think I start to see where the issue comes from as regards LUA_PATH... but then again who sets it?

**Comment by fgeorgatos** [comment]
relief, this is really fixed now via 7.7.23 - thanks! :

```
[fgeorgatos@nodeXYZ ~]$ ml --version

Modules based on Lua: Version 7.7.23  2018-03-20 11:24 -05:00
    by Robert McLay mclay@tacc.utexas.edu

[fgeorgatos@nodeXYZ ~]$ mkdir posix.lua
[fgeorgatos@nodeXYZ ~]$
[fgeorgatos@nodeXYZ ~]$ rmdir posix.lua
[fgeorgatos@nodeXYZ ~]$ ml --version

Modules based on Lua: Version 7.7.23  2018-03-20 11:24 -05:00
    by Robert McLay mclay@tacc.utexas.edu

[fgeorgatos@nodeXYZ ~]$ date
Wed Mar 21 04:57:16 PDT 2018
```

**Comment by fgeorgatos** [comment]
@rtmclay, fyi. that original read-only issue mentioned:
 I think it's not reproducible any more (i.e. I consider it as fixed)
